### DISCUSSION TITLE
How to detect user has just been authenticated?

### DISCUSSION THREAD

**Discussion by OlivierBOEL** [OpenIDC/mod_auth_openidc#1249] [Q&A]
Hello,

Apologies in advance if the answer is obvious...
I wonder how to detect user has just been authenticated i.e. request was unauthenticated (no or invalid session) and user has first been re-directed to OP, then after authentication to OIDCRedirectURI, then to requested resource; in other words, is there any environment variable that is set when a new session is created and variable modified (or unset) for existing session?

Thanks in advance,

Olivier

**Comment by jlbooker** [comment]
You'll want to look for the `REMOTE_USER` environment variable. You can control this with the `OIDCAuthRemoteUserClaim` [configuration directive](https://github.com/OpenIDC/mod_auth_openidc/blob/master/auth_openidc.conf#L504).

This variable will be set on every authenticated requested, not just when a new session is created. It'll be up to your application to detect that it was changed (i.e. a request was received with an empty `REMOTE_USER`, then a subsequent request where `REMOTE_USER` was set). You'll probably want to set your own application-specific cookie to indicate that the `REMOTE_USER` has been set AND that your application has done whatever initial processing it needs to do when the user first signs in.

**Reply by OlivierBOEL** [reply]
Hi, Jeremy,
Thanks for the feedback.
I was thinking (dreaming) of a more simple approach like an environment variable that would be set (or incremented) on every request; so, for initial (i.e. first authenticated) request, it would be set to 1. But, obviously, this is not available.
Thanks again,
Oliveir
